servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/AggregatedRequestsFilterTest.java

```diff
@@ -0,0 +1,213 @@
+import static io.servicetalk.concurrent.api.Publisher.from;
+import static io.servicetalk.concurrent.api.Single.failed;
+import static io.servicetalk.concurrent.api.Single.succeeded;
+import static io.servicetalk.http.api.HttpHeaderNames.TRANSFER_ENCODING;
+import static io.servicetalk.http.api.HttpHeaderValues.CHUNKED;
+import static io.servicetalk.http.api.HttpProtocolVersion.HTTP_1_1;
+import static io.servicetalk.http.api.HttpRequestMethod.GET;
+import static io.servicetalk.http.api.HttpResponseStatus.OK;
+import static io.servicetalk.http.api.HttpSerializationProviders.textSerializer;
+import static io.servicetalk.http.api.StreamingHttpRequests.newTransportRequest;
+import static io.servicetalk.http.netty.HttpServers.forPort;
+import static java.util.Objects.requireNonNull;
+import static org.junit.Assert.assertEquals;
+
+public class AggregatedRequestsFilterTest {
+
+    private static final StreamingHttpClientFilterFactory BODY_TRANSFORMER_FILTER =
+            new StreamingHttpClientFilterFactory() {
+        @Override
+        public StreamingHttpClientFilter create(final FilterableStreamingHttpClient client) {
+            return new StreamingHttpClientFilter(client) {
+                @Override
+                protected Single<StreamingHttpResponse> request(final StreamingHttpRequester delegate,
+                                                                final HttpExecutionStrategy strategy,
+                                                                final StreamingHttpRequest request) {
+                    BufferAllocator allocator = delegate.executionContext().bufferAllocator();
+                    request.transformPayloadBody(b -> b.concat(succeeded(allocator.fromAscii("red-pill"))));
+                    return super.request(delegate, strategy, request);
+                }
+            };
+        }
+    };
+
+    private static final StreamingHttpClientFilterFactory RAW_TRANSFORMER_FILTER =
+            new StreamingHttpClientFilterFactory() {
+                @Override
+                public StreamingHttpClientFilter create(final FilterableStreamingHttpClient client) {
+                    return new StreamingHttpClientFilter(client) {
+                        @Override
+                        protected Single<StreamingHttpResponse> request(final StreamingHttpRequester delegate,
+                                                                        final HttpExecutionStrategy strategy,
+                                                                        final StreamingHttpRequest request) {
+                            BufferAllocator allocator = delegate.executionContext().bufferAllocator();
+                            HttpHeadersFactory headersFactory = new DefaultHttpHeadersFactory(false, false);
+
+                            request.transformRawPayloadBody(b -> from(allocator.fromAscii("red-pill"),
+                                    headersFactory.newTrailers().add("fooA", "barA")));
```

Comment:
Discussed offline: `transformRawPayloadBody` should not add trailers. For trailers, `transform/transformRaw` methods should be used. Also, users are responsible for consuming the original publisher when they apply transformations.